bool has_result
bool success
string info
